### DISCUSSION TITLE
Shared maxForks=1 for a group of processes

### DISCUSSION THREAD

**Discussion by kwonsoobin** [nextflow-io/nextflow#3293] [Q&A]
Is there anyway to define a shared maxForks for multiple processes?

I have a pipeline of `input | A | B | C | D`. I want exactly one process instance to run at a time for `B` and `C`. For example, if `C` is running, I do not want `B` to run even if the input is different. `A` and `D` can run whenever. 

I tried assigning the same label to `B` and `C` and then setting `maxForks` to 1 for that label in my config file as shown below. This seems to set `maxForks` to 1 for each process and not for both processes combined.
```
process {
    withLabel: one_at_a_time {
        maxForks = 1
    }
}
```

Any suggestions? I don't think I can define maxForks or queueSize for a workflow. My last resort would be combine `B` and `C` into one process, which I want to avoid.

**Comment by jemunro** [comment]
One solution would be to add toList and flatMap between B and C.
E.g. `input | A | B | toList | flatMap | C | D`
That way C won't start until B is finished.
